src/main/java/athena/TaskList.java

```diff
@@ -0,0 +1,136 @@
+package athena;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    public static final String NO_FILTER = "";
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    private ArrayList<Task> getShortlistedTasks(String instance) {
+        ArrayList<Task> shortlistedTasks = new ArrayList<>();
+        for (Task task: tasks) {
+            if (task.getName().contains(instance)) {
+                shortlistedTasks.add(task);
+            }
+        }
+        return shortlistedTasks;
+    }
+
+    private void printAllTasks() {
+        for (Task task : tasks) {
+            System.out.println(task);
+        }
+    }
+
+    private void printTasksWithFilter(String importanceFilter) {
```

Comment:
Can call the printList() function from ui and pass it the filtered list (so tasklist will still handle the filtering)